modules/Bio/EnsEMBL/Production/Pipeline/PostGenebuild/LoadAppris.pm

```diff
@@ -115,28 +118,21 @@ sub run {
       $gene_cnt++;
       foreach my $transcript ( @{ $gene->get_all_Transcripts } ) {
         $transc_cnt++;
-
-        if (
-          exists $appris_results{ $gene->stable_id }{ $transcript->stable_id } )
-        {
+        if (exists $appris_results{ $gene->stable_id }{ $transcript->stable_id } ){
           # oh good, found this stable ID
           $stable_id_in_file++;
-
-          my %res =
-            $appris_results{ $gene->stable_id }{ $transcript->stable_id };
-
-          my $attrib = $attribs->{ $res{attrib_type_code} };
-
+          my $res = {};
+          $res = $appris_results{ $gene->stable_id }->{ $transcript->stable_id };
+          my $attrib = $attribs->{ $res->{attrib_type_code} };
           $aa->store_on_Transcript( $transcript, [
                                       Bio::EnsEMBL::Attribute->new(
                                                    -NAME  => $attrib->[2],
                                                    -CODE  => $attrib->[1],
-                                                   -VALUE => $res{attrib_value},
+                                                   -VALUE => $res->{attrib_value},
                                                    -DESCRIPTION => $attrib->[3]
-                                      ) ] )
-
-            print STDERR "  writing gene " . $gene->stable_id . " transcript" .
-            $transcript->stable_id . " APPRIS " . $res{attrib_type_code} . "\n";
+                                      ) ] );
+            print STDERR "  writing gene " . $gene->stable_id . " transcript " .
```

Comment:
I think this is probably better logged with `$self->warning(<text>)`, so that it ends up in the hive db's log_message table, as 'INFO', rather than as an error.

Reply:
@james-monkeyshines: This will actually create a lot of rows in the hive table (37776 for human alone...), this is why I didn't use $self->warning(). But this should not be STDERR, maybe I should use the logger instead so you will see it when debugging but it won't overflow the hive db.

Reply:
Ah, I see. Another option might be to only do this if `$self->debug` is set.

Reply:
Good point @james-monkeyshines, I will do this